### PR TITLE
Feature/#33673 add deadlines for worksheets

### PR DESCRIPTION
**Pull request by jtgoltz**
…ine to be set

### CODE REVIEW

File: forms/mumie_task_validator.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            if ($completionpass && (empty($data['gradepass']) || grade_floatval($data['gradepass']) == 0)) {
+                if (isset($data['completionpass'])) {
+                    $errors['completionpassgroup'] = get_string('gradetopassnotset', 'mumie');
+                } else {
+                    $errors['gradepass'] = get_string('gradetopassmustbeset', 'mumie');
+                }
+            }
+        }
+
+        if (self::has_duedate($data)) {
```
Comment (fsacha) [inline forms/mumie_task_validator.php:75]: `if (self::hasDueDate($data) && time() - $data['duedate'] > 0) {
        $errors['duedate'] = get_string('mumie_form_due_date_must_be_future', 'mod_mumie');
    }`
Is better than a nested if statement I think

File: lang/de/mumie.php [hunk 2]
```diff
@@ -115,6 +115,7 @@
 $string['mumie_form_wiki_link'] = 'Bitte besuchen Sie unser <a target="_blank" href="https://wiki.mumie.net/wiki/MUMIE-Moodle-integration-for-teachers">Wiki</a> für weiterführende Informationen!';
 $string['mumie_form_no_course_on_server'] = 'MUMIE-Server <b>{$a}</b> konnte nicht erreicht werden! Dieser Server wurde daher vorrübergehend aus der Auswahl entfernt.';
 $string['mumie_multi_problem_selector'] = 'Alternativ können Sie auch mehrere MUMIE Tasks gleichzeitig mit der Mehrfach-Aufgabenauswahl erstellen. Ziehen Sie dazu einfach die gewünschten Aufgaben in ihren Kurs.<br><br>Für eine Kurzanleitung und weitere Informationen besuchen Sie unser <a target="_blank" href="{$a}">Wiki</a>';
+$string['mumie_form_deadline_required_for_trigger_after_deadline'] = "Das ausgewählte Worksheet setzt eine Deadline vorraus";
```
Comment (fsacha) [inline lang/de/mumie.php:118]: voraus nicht vorraus :)

File: forms/mumie_task_validator.php (hunk 1 above)
Comment (jtgoltz) [reply forms/mumie_task_validator.php]: True. I just copy&pasted the old validation logic into this new class and didn't refactor it.

Changed

File: lang/de/mumie.php (hunk 2 above)
Comment (jtgoltz) [reply lang/de/mumie.php]: Good catch. Fixed

**Review by fsacha — APPROVED** [review]

**Pull request merged**